### PR TITLE
Dtsbps 462 service

### PR DESCRIPTION
**Pull request by gauravchawla73463**
### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java [hunk 1]
```diff
@@ -79,6 +80,24 @@ public ZipFileStatus getStatusFor(String zipFileName) {
...
+            List<ProcessEvent> events = eventRepo.findByZipFileName(zipFileName);
+            return new ZipFileStatus(
+                zipFileName,
+                envelopes.stream().map(this::mapEnvelope).collect(toList()),
+                events.stream().map(this::mapEvent).collect(toList())
+            );
+        }
+        return new ZipFileStatus(
+        "",
+         Collections.emptyList(),
```
Comment (apereverzin) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java:97]: Use static import

Comment (gauravchawla73463) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java]: done

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java [hunk 2]
```diff
@@ -79,6 +80,24 @@ public ZipFileStatus getStatusFor(String zipFileName) {
         return zipFileStatusList;
     }
 
+    public ZipFileStatus getStatusByCcdId(String ccdId) {
```
Comment (apereverzin) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java:83]: It should return `CcdIdStatus` extending `ZipFileStatus` with added `String ccdId` attribute. In this case the resulting JSon will be displaying the search criteria along with the result; this will be in line with searching by `fileName` - the search criteria is provided disregarding do we know the file or not.

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java [hunk 3]
```diff
@@ -79,6 +81,28 @@ public ZipFileStatus getStatusFor(String zipFileName) {
...
+            ZipFileStatus zipFileStatus = new ZipFileStatus(zipFileName,
+                                          envelopes.stream().map(this::mapEnvelope).collect(toList()),
+                                          events.stream().map(this::mapEvent).collect(toList()));
+            return new CcdIdStatus(
+                ccdId,
+                zipFileStatus
+            );
+        }
+        return new CcdIdStatus(
+        "",
```
Comment (apereverzin) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java:99]: Should be `ccdId`

Comment (gauravchawla73463) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java]: Yes, makes sense, it will return ccdid with empty zipfilestatus now.

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/model/out/zipfilestatus/ZipFileStatus.java [hunk 4]
```diff
@@ -9,15 +10,20 @@
     @JsonProperty("file_name")
```
Comment (mdayican) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/model/out/zipfilestatus/ZipFileStatus.java:10]: needs 
    @JsonInclude(JsonInclude.Include.NON_NULL)

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java (hunk 2 above)
Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java:85]: I think we do not need to extend.

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/model/out/zipfilestatus/ZipFileStatus.java (hunk 4 above)
Comment (gauravchawla73463) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/model/out/zipfilestatus/ZipFileStatus.java:11]: done

**Review by mdayican — COMMENTED** [review]
need to update existing  filename tests to assert ccdid=null

File: src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusServiceTest.java [hunk 5]
```diff
@@ -298,6 +299,96 @@ public void should_return_invalid_parameter_exception_for_dcn_less_than_6_chars(
...
+        String ccdId = "7452652475246754";
+        List<Envelope> envelopes = asList(envelope("A"), envelope("B"));
+        given(envelopes.get(0).getZipFileName()).willReturn("hello.zip");
+        given(envelopeRepo.findByCcdId(ccdId)).willReturn(envelopes);
+        given(eventRepo.findByZipFileName("hello.zip")).willReturn(events);
+        // when
+        ZipFileStatus result = service.getStatusByCcdId(ccdId);
+        // then
+        assertThat(result.ccdId).isEqualTo(ccdId);
+        assertThat(result.fileName).isEqualTo("hello.zip");
```
Comment (mdayican) [inline src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusServiceTest.java:319]: I think this should be null (not exist) if we are returning what is searched for

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java [hunk 6]
```diff
@@ -79,6 +82,25 @@ public ZipFileStatus getStatusFor(String zipFileName) {
...
+
+        List<Envelope> envelopes = envelopeRepo.findByCcdId(ccdId);
+        if (envelopes.size() > 0) {
+            String zipFileName = envelopes.get(0).getZipFileName();
+            List<ProcessEvent> events = eventRepo.findByZipFileName(zipFileName);
+
+            return new ZipFileStatus(zipFileName,
+                                     ccdId,
+                                     envelopes.stream().map(this::mapEnvelope).collect(toList()),
+                                     events.stream().map(this::mapEvent).collect(toList()));
```
Comment (apereverzin) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java:95]: these mappings (for envelopes and events) are duplicated 3 times in this class, this duplication should be eliminated

Comment (gauravchawla73463) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java]: its done

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java [hunk 7]
```diff
@@ -68,24 +64,41 @@ public ZipFileStatus getStatusFor(String zipFileName) {
...
             zipFileName ->
                 zipFileStatusList.add(
-                    new ZipFileStatus(
-                        zipFileName,
-                        envelopeRepo.findByZipFileName(zipFileName).stream().map(this::mapEnvelope).collect(toList()),
-                        eventRepo.findByZipFileName(zipFileName).stream().map(this::mapEvent).collect(toList())
-                    )
+                    getZipFileStatus(zipFileName, null,
+                                     envelopeRepo.findByZipFileName(zipFileName),
+                                     eventRepo.findByZipFileName(zipFileName))
```
Comment (mdayican) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java:69]: formatting

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java [hunk 8]
```diff
@@ -68,24 +64,41 @@ public ZipFileStatus getStatusFor(String zipFileName) {
...
+        if (envelopes.size() > 0) {
+            String zipFileName = envelopes.get(0).getZipFileName();
+            List<ProcessEvent> events = eventRepo.findByZipFileName(zipFileName);
+            return getZipFileStatus(null,ccdId,envelopes, events);
+        }
+        return getZipFileStatus(null, ccdId, emptyList(), emptyList());
+    }
+
+    private ZipFileStatus getZipFileStatus(String fileName, String ccdId,
+                                           List<Envelope> envelopes, List<ProcessEvent> events) {
```
Comment (mdayican) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java:87]: formatting

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java [hunk 9]
```diff
@@ -68,24 +64,41 @@ public ZipFileStatus getStatusFor(String zipFileName) {
...
+        return new ZipFileStatus(fileName,
+                          ccdId,
+                          envelopes.stream().map(this::mapEnvelope).collect(toList()),
+                          events.stream().map(this::mapEvent).collect(toList()));
```
Comment (mdayican) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java:91]: formatting

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java [hunk 10]
```diff
@@ -46,19 +47,14 @@ public ZipFileStatusService(
...
-        );
+        return getZipFileStatus(zipFileName, null, envelopes, events);
     }
 
     public List<ZipFileStatus> getStatusByDcn(String documentControlNumber) throws InvalidParameterException {
 
         if (documentControlNumber.length() < MIN_LENGTH) {
             log.error("Exception in Search by DCN error: DCN number specified is less than"
-                      + MIN_LENGTH + " characters in length.");
+                          + MIN_LENGTH + " characters in length.");
```
Comment (mdayican) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java:57]: MIN_LENGTH should  pass as parameter.

**Review by mdayican — APPROVED** [review]
